cmd/minikube/cmd/service.go

```diff
@@ -68,11 +71,16 @@ var serviceCmd = &cobra.Command{
 		if !cluster.IsMinikubeRunning(api) {
 			os.Exit(1)
 		}
-		err = service.WaitAndMaybeOpenService(api, namespace, svc,
+		urlString, err := service.WaitForService(api, namespace, svc,
 			serviceURLTemplate, serviceURLMode, https, wait, interval)
 		if err != nil {
 			exit.WithError("Error opening service", err)
 		}
+
+		if len(urlString) != 0 {
```

Comment:
`WaitAndMaybeOpenService` had the following logic for opening a URL:

`if urlMode || !isHTTPSchemedURL {`

Make sure your PR preserves the same behavior as before.

Reply:
The code that you are referring to is inside pkg/minikube/service/service.go and the only changes made was to send back the array of urls